### ISSUE TITLE
twisted.test.iosim.FakeTransport incorrectly implements getHost and getPeer

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6368]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6368|
|Type|enhancement|
|Created|2013-03-14 00:41:49Z|
|Branch|https://github.com/twisted/twisted/tree/iosim-transport-addresses-6368|

These should return an `IAddress` provider, not a tuple.  As different tests may want to exercise handling of different transport types / address types, the host and peer should probably be parameters as they are for StringTransport.

<details><summary>Searchable metadata</summary>

```
trac-id__6368 6368
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_iosim_transport_addresses_6368 branches/iosim-transport-addresses-6368
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1363221709000000 1363221709000000
changetime__1363645873000000 1363645873000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [37501]) Branching to 'iosim-transport-addresses-6368'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/iosim-transport-addresses-6368)

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_3"></a>|@jerub set owner to @exarkun|
|-|-|

I have no strong opinion about the spaces around methods, but you should be aware that you added spaces between __repr__ and other methods, but didn't change anything else.

Remove this comment from iosim.py:          # XXX: According to ITransport, this should return an IAddress!

Important lint warnings:
C:140: Line too long (82/80)
C:333: Line too long (90/80)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @jerub|
|-|-|

Fixed those.  New build currently in progress, [results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/iosim-transport-addresses-6368).  Also fixed a failure in the AMP tests.

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_5"></a>|@jerub set owner to @exarkun|
|-|-|

Thanks, twistedchecker isn't too happy, but it looks like a bug in buildbot.

Please merge when you're ready.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [37636]) Merge iosim-transport-addresses-6368

Author: exarkun
Reviewer: jerub
Fixes: [#6368](https://github.com/twisted/twisted/issues/6368)

Change the in-memory transport implementation in `twisted.test.iosim` to return
`IAddress` providers from its `getHost` and `getPeer` methods (instead of tuples).
